pkg/apis/core/v1/defaults.go

```diff
@@ -215,6 +198,17 @@ func SetDefaults_Pod(obj *v1.Pod) {
 				}
 			}
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
+			isRestartableInitContainer := false
+			c := obj.Spec.InitContainers[i]
+			if c.RestartPolicy != nil && *c.RestartPolicy == v1.ContainerRestartPolicyAlways {
+				isRestartableInitContainer = true
+			}
+			if isRestartableInitContainer && (c.Resources.Requests != nil || c.Resources.Limits != nil) {
+				// For restartable init containers, set resize restart policy to default value (NotRequired), if not specified.
+				setDefaultResizePolicy(&obj.Spec.InitContainers[i])
+			}
+		}
```

Comment:
```suggestion
		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
			c := obj.Spec.InitContainers[i]
			if podutil.IsRestartableInitContainer(&c) && (c.Resources.Requests != nil || c.Resources.Limits != nil) {
				// For restartable init containers, set resize restart policy to default value (NotRequired), if not specified.
				setDefaultResizePolicy(&obj.Spec.InitContainers[i])
			}
		}
```

Reply:
Thanks for catching this. I have updated the code.

Reply:
Actually after looking again, I was getting the import cycle error. So that's why I didn't use the podutil helper function here. See this revert commit: https://github.com/kubernetes/kubernetes/pull/128367/commits/26e6d2934e9010dc10b2963fddb432df43a51676